haystack/indexing/io.py

```diff
@@ -24,23 +23,28 @@ def write_documents_to_db(document_dir, clean_func=None, only_empty_db=False):
 
     # check if db has already docs
     if only_empty_db:
-        n_docs = db.session.query(Document).count()
+        n_docs = len(datastore.get_all_documents())
```

Comment:
This might get expensive for large collections of docs. I would suggest having a dedicated method `datastor.count_docs()` or similar.